ArduCopter/inertia.cpp

```diff
@@ -20,7 +20,7 @@ void Copter::read_inertia()
     // current_loc.alt is alt-above-home, converted from inertial nav's alt-above-ekf-origin
     const int32_t alt_above_origin_cm = inertial_nav.get_position_z_up_cm();
     current_loc.set_alt_cm(alt_above_origin_cm, Location::AltFrame::ABOVE_ORIGIN);
-    if (!ahrs.home_is_set() || !current_loc.change_alt_frame(Location::AltFrame::ABOVE_HOME)) {
```

Comment:
Need to be careful though because get_alt_cm() can cause a panic on SITL.

Reply:
Hmmm.  My reading of the flow was that we checked for that on the object you call `change_alt_frame` on - but we don't call `get_alt_cm` on home, we grab the alt directly from the object (after checking home_is_set, even)

Did I misread the flow?

Reply:
We specifically added that check after a SITL failure https://github.com/ArduPilot/ardupilot/pull/13489 so I'm concerned that we're letting that failure come back.

Reply:
> We specifically added that check after a SITL failure #13489 so I'm concerned that we're letting that failure come back.

Ah!  Great memory :-)

I think the original fix was actually not quite right.  I think my reason was sound, but it's actually `current_location` which can be all-zeroes.  I've changed this patch to add that check.

We discussed on the call on Monday about having Home be able to be set regardless of origin being set - this would be an important fix for that scenario.